File: app/src/main/java/net/nashihara/naroureader/adapters/RankingFragmentPagerAdapter.java
package net.nashihara.naroureader.adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

public class RankingFragmentPagerAdapter extends FragmentPagerAdapter {
    private static final String TAG = RankingFragmentPagerAdapter.class.getSimpleName();

    private Fragment[] fragments;
    private String[] titles;

    public RankingFragmentPagerAdapter(FragmentManager fm, Fragment[] fragments, String[] titles) {
        super(fm);
        this.fragments = fragments;
        this.titles = titles;
    }

    @Override
    public Fragment getItem(int position) {
        if (position >= fragments.length) {
            return null;
        } else {
            return fragments[position];
        }
    }

    @Override
    public int getCount() {
        return 4;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if (position >= titles.length) {
            return null;
        } else {
            return titles[position];
        }
    }
}

File: app/src/main/java/net/nashihara/naroureader/adapters/RankingFragmentPagerAdapter.kt
package net.nashihara.naroureader.adapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

class RankingFragmentPagerAdapter(
        fm: FragmentManager,
        private val fragments: Array<Fragment>,
        private val titles: Array<String>) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment? {
        if (position >= fragments.size) {
            return null
        } else {
            return fragments[position]
        }
    }

    override fun getCount(): Int {
        return fragments.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        if (position >= titles.size) {
            return null
        } else {
            return titles[position]
        }
    }
}
